### PR TITLE
MDCT-265 Setup Automated Deployments

### PR DESCRIPTION
**Pull request by bodnarbm**
## Description

These changes should enable both dev branches being automatically deployments for review and validation and handling deployments for our main, val, and production environments.

Changes that were required for this to work
- Switched to using OIDC for AWS authentication instead of static user credentials for the infrastructure deployments.
- Updates to branch naming references to match this repo's branch name conventions for the environments.
- Updates to cypress test setup to work within CI and target deployed infrastructure.
- Fix for CI builds of ui-src (addresses autoprefixer warning that becomes error on CI builds)

[MDCT-265](https://qmacbis.atlassian.net/browse/MDCT-265)

### How to test
CI successfully deploys the dev branch and cypress tests pass.

### Changed Dependencies
Cypress dependencies moved from root package.json to the cypress package.json. This is to allow the cypress github action to be independent.

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/patches/@cmsgov+design-system+2.12.0.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@cmsgov/design-system/dist/scss/components/_Pagination.scss b/node_modules/@cmsgov/design-system/dist/scss/components/_Pagination.scss
+index 10da57c..bdf4b0d 100644
+--- a/node_modules/@cmsgov/design-system/dist/scss/components/_Pagination.scss
++++ b/node_modules/@cmsgov/design-system/dist/scss/components/_Pagination.scss
+@@ -9,7 +9,7 @@ $pagination-overflow-color: $color-gray;
+ $pagination-page-count-color: $color-gray;
+ 
+ .ds-c-pagination {
+-  align-items: end;
++  align-items: flex-end;
```
[the remark this answers is not in the document]
Comment (braxex) [reply services/ui-src/patches/@cmsgov+design-system+2.12.0.patch:10]: I'm all for that. It's an annoying warning.

**Review by braxex — APPROVED** [review]

**Review by gmrabian — APPROVED** [review]
lgtm

File: services/ui-src/patches/@cmsgov+design-system+2.12.0.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (bodnarbm) [reply services/ui-src/patches/@cmsgov+design-system+2.12.0.patch]: Upgrading this now.

[the remark this answers is not in the document]
Comment (bodnarbm) [reply services/ui-src/patches/@cmsgov+design-system+2.12.0.patch:10]: Never mind. The upgrade causes some test failures, so will likely need to be done in it's own scope of work.

**Pull request merged**